addon/lint/lint.js

```diff
@@ -187,7 +187,7 @@
   CodeMirror.defineOption("lint", false, function(cm, val, old) {
     if (old && old != CodeMirror.Init) {
       clearMarks(cm);
-      if (!state.options.lintOnChange !== false)
+      if (cm.state.lint.options.lintOnChange === undefined || cm.state.lint.options.lintOnChange === true)
```

Comment:
> Am I missing anything?

Yes, you are. The precedence of `!` is higher than that of `!==`, so you're comparing `(!a.lintOnChange)` to `false`.